flytekit/models/launch_plan.py

```diff
@@ -233,14 +233,27 @@ def from_flyte_idl(cls, pb2):
         :param flyteidl.admin.launch_plan_pb2.LaunchPlanSpec pb2:
         :rtype: LaunchPlanSpec
         """
+        auth_role = None
+        # First check the newer field, auth_role.
+        if pb2.auth_role is not None and (
+            pb2.auth_role.HasField("assumable_iam_role") or pb2.auth_role.HasField("kubernetes_service_account")
```

Comment:
Take a look [at this](https://github.com/googleapis/google-cloud-python/issues/1402) regarding HasField and proto3